lib/pbench/test/unit/server/test_requests.py

```diff
@@ -170,3 +172,204 @@ def test_upload(client, pytestconfig, caplog):
         )
         for record in caplog.records:
             assert record.levelname not in ("WARNING", "ERROR", "CRITICAL")
+
+
+@pytest.fixture
+def query_helper(client, requests_mock):
+    """
+    query_helper Help controller queries that want to interact with a mocked
+    Elasticsearch service.
+
+    This is a fixture which exposes a function of the same name that can be
+    used to set up and validate a mocked Elasticsearch query with a JSON
+    payload and an expected status.
+
+    Parameters to the mocked Elasticsearch POST are passed as keyword
+    parameters: these can be any of the parameters supported by the
+    request_mock post method. The most common are 'json' for the JSON
+    response payload, and 'exc' to throw an exception.
+
+    :return: the response object for further checking
+    """
+
+    def query_helper(payload, expected_index, expected_status, **kwargs):
+        requests_mock.post(re.compile(f'{client.config["ES_URL"]}'), **kwargs)
+        response = client.post(
+            f"{client.config['REST_URI']}/controllers/list", json=payload
+        )
+        assert requests_mock.last_request.url == (
+            client.config["ES_URL"]
+            + expected_index
+            + "/_search?ignore_unavailable=true"
+        )
+        assert response.status_code == expected_status
+        return response
+
+    return query_helper
+
+
+class TestQueryControllers:
+    """
+    Unit testing for resources/QueryControllers class.
+
+    In a web service context, we access class functions mostly via the
+    Flask test client rather than trying to directly invoke the class
+    constructor and `post` service.
+    """
+
+    def build_index(self, client, dates):
+        """
+        build_index Build the index list for query
+
+        Args:
+            dates (iterable): list of date strings
+        """
+        prefix = client.config["PREFIX"]
+        idx = prefix + ".v6.run-data."
```

Comment:
`v6` seems like a hard-coded value, is there a way we could avoid this without too much hassle?

Reply:
Yeah; it's hardcoded in `query_controllers.py`, too: check out the large `TODO` at line 200.

The Dashboard handles this as a config file option. While we could do that too, ideally I'd like to get it from the template files when we first load them... e.g., store the version for each "root" index name in a persistent DB (Redis?). That'll be relatively straightforward when we go to microservices.

Another option would be for `indexer.py` to write a JSON file somewhere e.g. under {pbench-top-dir} to store the mapping, which other components could reference. In any case, unless you're going to strong-arm my conscience right now, my thought was to create an issue and defer that.